boot/bootutil/src/loader.c

```diff
@@ -2458,11 +2507,24 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
         }
 
 #ifdef MCUBOOT_VALIDATE_PRIMARY_SLOT
-        rc = boot_validate_slot(state, BOOT_PRIMARY_SLOT, NULL);
-        if (rc != 0) {
-            rc = BOOT_EBADIMAGE;
-            goto out;
-        }
+#ifdef PM_S1_ADDRESS
+	/* Patch needed for NCS. If secure boot is enabled, then mcuboot
+	 * will be stored in either partition S0 or S1. Image 1 primary
+	 * will point to the 'other' Sx partition. Hence, image 1 primary
+	 * does not contain a valid image until mcuboot has been upgraded.
+	 * Note that B0 will perform validation of the active mcuboot image,
+	 * so there is no security lost by skipping this check for image 1
+	 * primary.
+	 */
+	if (BOOT_CURR_IMG(state) == 0)
```

Comment:
This code block looks strange after preprocessor will cut it for  PM_S1_ADDRESS

Reply:
It should still be valid (empty { } block provides no semantics, but is legal.
If I do this a different way the diff will be bigger.

Reply:
ping @nvlsianpu